src/basic/fileio.c

```diff
@@ -375,8 +375,10 @@ int write_string_file_full(
         return 0;
 
 fail:
-        if (call_label_ops_post)
-                (void) label_ops_post(fd >= 0 ? fd : dir_fd, fd >= 0 ? NULL : fn, made_file);
+        if (call_label_ops_post) {
```

Comment:
I would prefer to unify the two label_ops_post() callsites, i.e. instead of goto fail immediately after fd < 0, move the call_label_ops_post branch in between. Feels more readable that way, plus it's already how we treat mac_selinux_* funcs

Reply:
Moreover we avoid duplicated calls in the case of write_string_file_atomic_at